sw/device/silicon_creator/lib/base/sec_mmio.c

```diff
@@ -65,28 +57,30 @@ void sec_mmio_init(sec_mmio_shutdown_handler cb) {
     sec_mmio_ctx.addrs[i] = UINT32_MAX;
     sec_mmio_ctx.values[i] = UINT32_MAX;
   }
+  uint32_t check = kSecMmioValZero ^ launder32(sec_mmio_ctx.last_index);
+  check ^= launder32(sec_mmio_ctx.write_count);
+  check ^= launder32(sec_mmio_ctx.check_count);
+  check ^= launder32(sec_mmio_ctx.expected_write_count);
+  HARDENED_CHECK_EQ(check, kSecMmioValZero);
 }
 
-void sec_mmio_next_stage_init(sec_mmio_shutdown_handler cb) {
-  sec_mmio_shutdown_cb = cb;
+void sec_mmio_next_stage_init(void) {
   sec_mmio_ctx.check_count = 0;
   for (size_t i = sec_mmio_ctx.last_index; i < ARRAYSIZE(sec_mmio_ctx.addrs);
        ++i) {
     sec_mmio_ctx.addrs[i] = UINT32_MAX;
     sec_mmio_ctx.values[i] = UINT32_MAX;
   }
+  HARDENED_CHECK_EQ(sec_mmio_ctx.check_count, 0);
 }
 
 uint32_t sec_mmio_read32(uint32_t addr) {
   uint32_t value = abs_mmio_read32(addr);
   uint32_t masked_value = value ^ kSecMmioMaskVal;
 
   upsert_register(addr, masked_value);
+  HARDENED_CHECK_EQ((abs_mmio_read32(addr) ^ kSecMmioMaskVal), masked_value);
```

Comment:
Parentheses are not needed here (if they are, it's a bug).

Reply:
Done